store-implementation/federated-store/src/test/java/uk/gov/gchq/gaffer/federatedstore/FederatedStoreGetElementsWithinSetTest.java

```diff
@@ -0,0 +1,273 @@
+import static uk.gov.gchq.gaffer.federatedstore.FederatedStoreTestUtil.resetForFederatedTests;
+import static uk.gov.gchq.gaffer.store.TestTypes.BOOLEAN_TYPE;
+import static uk.gov.gchq.gaffer.store.TestTypes.DIRECTED_EITHER;
+import static uk.gov.gchq.gaffer.store.TestTypes.STRING_TYPE;
+
+import java.util.HashSet;
+import java.util.LinkedHashSet;
+import java.util.Set;
+
+public class FederatedStoreGetElementsWithinSetTest {
+    private static final AccumuloProperties ACCUMULO_PROPERTIES = AccumuloProperties.loadStoreProperties(StreamUtil.openStream(FederatedStoreGetElementsWithinSetTest.class, "properties/singleUseAccumuloStore.properties"));
+    private static final User USER = new User();
+    private static final Set<EntityId> SEEDS = new LinkedHashSet<>();
+    private static final String GRAPH_IDS = String.format("%s,%s", GRAPH_ID_A, GRAPH_ID_B);
+    private static final View EDGE_VIEW = new View.Builder().edge(GROUP_BASIC_EDGE).build();
+    private static final View ENTITY_VIEW = new View.Builder().entity(GROUP_BASIC_ENTITY).build();
+
+    private Graph federatedGraph;
+
+    @AfterAll
+    public static void tearDownCache() {
+        resetForFederatedTests();
+    }
+
+    @BeforeEach
+    public void setUp() throws Exception {
+        resetForFederatedTests();
+        FederatedStoreProperties federatedStoreProperties = getFederatedStorePropertiesWithHashMapCache();
+        for (int i = 0; i < 300; i++) {
+            SEEDS.add(new EntitySeed("A" + i));
+        }
+
+        federatedGraph = new Graph.Builder()
+                .config(new GraphConfig.Builder()
+                        .graphId(GRAPH_ID_TEST_FEDERATED_STORE)
+                        .build())
+                .addStoreProperties(federatedStoreProperties)
+                .build();
+
+        federatedGraph.execute(new AddGraph.Builder()
+                .graphId(GRAPH_ID_A)
+                .storeProperties(ACCUMULO_PROPERTIES)
+                .schema(getSchema())
+                .build(), USER);
+
+        federatedGraph.execute(new AddGraph.Builder()
+                .graphId(GRAPH_ID_B)
+                .storeProperties(ACCUMULO_PROPERTIES)
+                .schema(getSchema())
+                .build(), USER);
+
+        for (int i = 300; i >= 0; i--) {
+            addEntities("A" + i, GRAPH_ID_A);
+            addEntities("A" + i, GRAPH_ID_B);
+        }
+
+        // Add 6 edges total - should all have src and dest in different batches
+        addEdges("A244", "A87", GRAPH_ID_A);
+        addEdges("A168", "A110", GRAPH_ID_A);
+        addEdges("A56", "A299", GRAPH_ID_A);
+        addEdges("A297", "A193", GRAPH_ID_B);
+        addEdges("A15", "A285", GRAPH_ID_B);
+        addEdges("A1", "A52", GRAPH_ID_B);
+    }
+
+    @Test
+    public void shouldReturnOnlyEdgesWhenViewContainsNoEntitiesFromSubGraphsWithFederatedOperation() throws Exception {
```

Comment:
Not major but you can probably leave out the `FromSubGraphsWithFederatedOperation` from these test names you're already in a federated test class its probably implied.